drivers/sensor/qdec_nrfx/qdec_nrfx.c

```diff
@@ -74,14 +74,16 @@ static int qdec_nrfx_sample_fetch(const struct device *dev,
 	return 0;
 }
 
+#define QDEC_STEPS DT_INST_PROP(0, steps)
+
 static int qdec_nrfx_channel_get(const struct device *dev,
 				 enum sensor_channel  chan,
 				 struct sensor_value *val)
 {
 	struct qdec_nrfx_data *data = &qdec_nrfx_data;
 	unsigned int key;
 	int32_t acc;
-	const int32_t steps = DT_INST_PROP(0, steps);
+	const int32_t steps = QDEC_STEPS;
```

Comment:
Alternatively, you could avoid the macro by using:
```suggestion
	enum { steps = DT_INST_PROP(0, steps) };
```

Reply:
Will the enum work with th build assert then though?

Reply:
It will. It's a "more constant" value than a `const` variable. :wink:

Reply:
Makes sense. But it’s not an enum originally right? So it’d be a bit misleading to use it?

Reply:
Why misleading? It's just a named value. And it is of integer type so it can fully replace that `int32_t` here. Perhaps its name could be changed to uppercase to emphasize that it is a constant.

Reply:
Using `enum` here would indeed be a very strange and unconventional choice, maybe even wrong depending on how you look at it.

Macro sounds reasonable to me (would be nicer if it were moved up to the beginning of the file with other defines). Also, could get rid of the redundant local `const int32_t steps` entirely and just use `QDEC_STEPS` everywhere.

Reply:
I agree with @stephanosio here. It's not an `enum` in the sense that it is not an enumerated value, instead it is a count, an integer. I feel it's not a clean map to an `enum`.